pkg/orgs_fetcher/orgs_fetcher.go

```diff
@@ -0,0 +1,93 @@
+package orgs_fetcher
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"code.cloudfoundry.org/lager"
+	cfclient "github.com/cloudfoundry-community/go-cfclient"
+)
+
+type OrgsStore interface {
+	GetOrgs() []cfclient.Org
+}
+
+type OrgsFetcher struct {
+	Orgs     []cfclient.Org
+	schedule time.Duration
+	logger   lager.Logger
+	cfClient cfclient.CloudFoundryClient
+	mu       sync.Mutex
+}
+
+func NewOrgsFetcher(
+	schedule time.Duration,
+	logger lager.Logger,
+	cfClient cfclient.CloudFoundryClient,
+) *OrgsFetcher {
+	logger = logger.Session("orgs-fetcher")
+	return &OrgsFetcher{
+		schedule:    schedule,
+		logger:      logger,
+		cfClient:    cfClient,
+	}
+}
+
+func (f *OrgsFetcher) GetOrgs() []cfclient.Org {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	return f.Orgs
+}
+
+func (f *OrgsFetcher) Run(ctx context.Context) error {
+	lsession := f.logger.Session("run")
```

Comment:
Can we make this something like `log_session` / `loggerSession`, just a touch more descriptive.